docs/design-documents/features/storage/SecureStore/SecureStore_design.md

```diff
@@ -0,0 +1,441 @@
+# SecureStore in Mbed OS
+
+- [SecureStore in Mbed OS](#securestore-in-mbed-os)
+    + [Revision history](#revision-history)
+- [Introduction](#introduction)
+    + [Overview and background](#overview-and-background)
+    + [Requirements and assumptions](#requirements-and-assumptions)
+- [System architecture and high-level design](#system-architecture-and-high-level-design)
+  * [Design basics](#design-basics)
+    + [Data layout](#data-layout)
+    + [Basic implementation concepts](#basic-implementation-concepts)
+- [Detailed design](#detailed-design)
+    + [Class header](#class-header)
+    + [Important data structures](#important-data-structures)
+    + [Initialization and reset](#initialization-and-reset)
+    + [Core APIs](#core-apis)
+    + [Incremental set APIs](#incremental-set-apis)
+    + [Key iterator APIs](#key-iterator-apis)
+- [Usage scenarios and examples](#usage-scenarios-and-examples)
+    + [Standard usage of the class](#standard-usage-of-the-class)
+- [Other information](#other-information)
+    + [Open issues](#open-issues)
+
+
+### Revision history
+
+| Revision 	| Date           	| Authors                                                	| Mbed OS version 	| Comments         	|
+|----------	|----------------	|--------------------------------------------------------	|-----------------	|------------------	|
```

Comment:
Regarding get and get_info functions: If record is not found we need to check also if they exist in rbp_kvstore

Reply:
And do what if we find them there?

Reply:
return authentication error

Reply:
OK. Do we remove the key in RBP storage?

Reply:
We  do not remove. We need the hint in RBP that at some time the file existed in normal KVSTore